ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs

```diff
@@ -79,9 +83,21 @@ type EraCrypto era = Crypto era
 -- needed to determine the hard fork point. In the future this should be
 -- replaced with an appropriate API - see
 -- https://github.com/input-output-hk/ouroboros-network/issues/2890
+--
+-- TODO Currently we include the constraint @SL.AdditionalGenesisConfig era ~
+-- ()@. When we fork to Alonzo we will need additional genesis config
+-- information.
 class ( SL.ShelleyBasedEra era
       , State (LC.EraRule "PPUP" era) ~ SL.PPUPState era
       , Default (State (LC.EraRule "PPUP" era))
+      , HasField "_maxBHSize" (LC.PParams era) Natural
```

Comment:
I'm somewhat hesitant to re-use GHC's infrastructure here directly, since I don't think of it as the most stable interface.

Did ledger consider introducing their own mediating class for this? ... I suppose we could just introduce our own `HasField`-shaped class later if GHC ends up changing directions.

Reply:
No, we use precisely `HasField`